Frontend_Web_Development_React_Angular_Vue/Vue/Week 2/computed_properties.md

```diff
@@ -0,0 +1,160 @@
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Document</title>
+    <!-- We are using a CDN to use Vue.js -->
+    <script src="https://cdn.jsdelivr.net/npm/vue@2.6.14/dist/vue.js"></script>
+</head>
+<body>
+    <div id = "content">
+        <h1>A Random number from 1-1000</h1>
+        <!-- These two h3 tags, will display a number that is returned from a method -->
+        <h3>Random number 1 from a method: {{ methodNumber() }}</h3>
+        <h3>Random number 2 from a method: {{ methodNumber() }}</h3>
+        <!-- These two h3 tags, will display a number that is returned from a computed property -->
+        <h3>Random number 1 from a computed property: {{ computedNumber }}</h3>
+        <h3>Random number 1 from a computed property: {{ computedNumber }}</h3>
+    </div>
+
+    <script>
+        let vueInstance = new Vue({
+            //We are targeting the div element with the ID 'content'
+            el: '#content',
+            //This is a method, that returns a random number
+            methods: {
+                methodNumber: function(){
+                return Math.floor(Math.random()*1000);
+            }},
+            //This is a computed property that returns a random number. 
+            computed: {
+                computedNumber: function(){
+                return Math.floor(Math.random()*1000);
+            }},
+        }
+        )
+    </script>
+</body>
+</html>
+```
+This code renders the following web page in the browser:
+
+![](./assets/computedProp.png)
+
+Notice how the random numbers from methods are different, but the ones from computed properties are always the same. No matter how many time the computed property is called it would instantly return the previously computed value and will not run the function again.
+
+# Basic Implementation:
+Let us consider this basic implementation of computed properties. This webpage, takes in two numerical inputs from the user and displays a product of those numbers. Notice how computed properties are triggered when the value in the input field is changed (because it is a reactive dependency).
+
+```html
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Document</title>
+    <!-- We are using a CDN to use Vue.js -->
+    <script src="https://cdn.jsdelivr.net/npm/vue@2.6.14/dist/vue.js"></script>
+</head>
+<body>
+    <div id = "demo" >
+        <!-- This div container has two input fields. The value in these input fields will be
+        assigned to the numberOne and numberTwo data properties in the vue instance.  -->
+        <div style="display: flex; flex-direction: row;justify-content: space-between; padding: 200px;">
+            <input style="width: 33%; height:50px;" type="number" v-model = "numberOne" placeholder="x">
```

Comment:
As binding is not covered yet. So please mention that.
"We are using 2-way binding with the help of v-model, which we will cover later in this Documentation"

Reply:
Please make this change. Then we are good to go with your PR.